torchrl/objectives/dreamer.py

```diff
@@ -49,6 +51,44 @@ class DreamerModelLoss(LossModule):
             Default: False.
     """
 
+    @dataclass
+    class _AcceptedKeys:
+        """Maintains default values for all configurable tensordict keys.
+
+        This class defines which tensordict keys can be set using '.set_keys(key_name=key_value)' and their
+        default values
+
+        Attributes:
+        ------------
+        reward : NestedKey
+            The reward is expected to be in the tensordict key ("next", reward). Defaults to ``"reward"``.
+        true_reward : NestedKey
+            The `true_reward` will be stored in the tensordict key ("next", true_reward). Defaults to ``"true_reward"``.
+        prior_mean : NestedKey
+            The prior mean is expected to be in the tensordict key ("next", prior_mean). Defaults to ``"prior_mean"``.
+        prior_std : NestedKey
+            The prior mean is expected to be in the tensordict key ("next", prior_mean). Defaults to ``"prior_mean"``.
+        posterior_mean : NestedKey
+            The posterior mean is expected to be in the tensordict key ("next", prior_mean). Defaults to ``"posterior_mean"``.
+        posterior_std : NestedKey
+            The posterior std is expected to be in the tensordict key ("next", prior_mean). Defaults to ``"posterior_std"``.
+        pixels : NestedKey
+            The pixels is expected to be in the tensordict key ("next", pixels). Defaults to ``"pixels"``.
+        reco_pixels : NestedKey
+            The reconstruction pixels is expected to be in the tensordict key ("next", reco_pixels). Defaults to ``"reco_pixels"``.
+        """
+
+        reward: NestedKey = "reward"
+        true_reward: NestedKey = "true_reward"
+        prior_mean: NestedKey = "prior_mean"
+        prior_std: NestedKey = "prior_std"
+        posterior_mean: NestedKey = "posterior_mean"
+        posterior_std: NestedKey = "posterior_std"
```

Comment:
Not 100% sure these guys belong here.
It's true that they are input to the loss module.
On the other hand, they are specific to our implementation of dreamer: if one is to change their names, it is likely that that person is also changing a lot of other things in the algo (ie it is a highly integrated feature).
Happy to keep them as part of this PR though, too many keys is better than too few.

Reply:
I had the same thought, but left them in because also for some other tensordict keys in other loss modules the same is true. In the end it felt more consistent to make all input keys configurable, especially it comes with no cost for the user when they just want the default behaviour.